spec-dtslint/operators/publish-spec.ts

```diff
@@ -0,0 +1,16 @@
+import { of, Observable } from 'rxjs';
+import { publish } from 'rxjs/operators';
+
+it('should support empty parameter', () => {
+  // Typescript limitations - inference only happen in one direction from operator to pipe
+  const a = of(1, 2, 3).pipe(publish()); // $ExpectType Observable<any>
+});
+
+it('should infer correctly with parameter', () => {
+  const a = of(1, 2, 3).pipe(publish((x: Observable<number>) => x)); // $ExpectType Observable<number>
+  const b = of('a', 'b', 'c').pipe(publish((x: Observable<string>) => x)); // $ExpectType Observable<string>
```

Comment:
I don't think the parameter types should be specified here, leave them to be inferred: `publish(x => x)`.

Reply:
@cartant I have removed the types for selectors. Thanks :)